restcomm/restcomm.mgcp/src/main/java/org/restcomm/connect/mgcp/MockMediaGateway.java

```diff
@@ -400,6 +439,26 @@ private void modifyConnection(final Object message, final ActorRef sender) {
     private void deleteConnection(final Object message, final ActorRef sender) {
         final ActorRef self = self();
         final DeleteConnection dlcx = (DeleteConnection) message;
+        if (dlcx.getConnectionIdentifier() == null) {
+            connEndpointMap.values().removeAll(Collections.singleton(dlcx.getEndpointIdentifier().getLocalEndpointName()));
+            monitoringService.tell(new MgcpConnectionDeleted(null, dlcx.getEndpointIdentifier().getLocalEndpointName()), self());
+            monitoringService.tell(new MgcpEndpointDeleted(dlcx.getEndpointIdentifier().getLocalEndpointName()), self());
+            if (logger.isInfoEnabled()) {
+                String msg = String.format("Endpoint deleted %s", dlcx.getEndpointIdentifier().getLocalEndpointName());
+                logger.info(msg);
+            }
+        } else {
+            connEndpointMap.remove(dlcx.getConnectionIdentifier().toString());
```

Comment:
same here, can we save the result of this operation for logging later?

Reply:
Sure as above